### PR TITLE
Adding myst-parser to jedi-tools-env

### LINKED ISSUE #234: Add Sphinx MyST-parser to spack and the to spack-stack
The `jedi-tools-env` needs package `myst-parser` for Sphinx to build the JEDI documentation (not needed for spack -stack documentation). This package doesn't exist yet in spack, therefore need to add it first in spack, and then make any necessary updates in spack-stack.

### PR DESCRIPTION
**Pull request by srherbener**
### Summary

Update submodule commit hash to point to the feature branch for https://github.com/JCSDA/spack/pull/305

### Testing

- CI testing
- @ashley314 and I both manually tested on our Mac. This covers intel base and arm based macs. https://github.com/JCSDA/spack/pull/305 worked successfully on both platforms.

### Applications affected

jedi-tools-env (this update is for building the html in the jedi-docs repo)

### Systems affected

Any platform where jedi-docs is being used - primarily mac laptops

### Dependencies

- [x] merge after https://github.com/JCSDA/spack/pull/305

### Issue(s) addressed

Fixes #234

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: .gitmodules
Comment (srherbener) [inline .gitmodules]: @ashley314 the change here is temporary and is done to help CI testing find your PRs feature branch. I named the feature branch in this PR the same as yours but that isn't necessary.

The idea here is:
- Once testing passes here, merge your PR in spack
- Then revert these changes (in .gitmodules)
- Then update the submodule commit hash to the one after your PR was merged
- Then merge this PR (once tests pass and reviewers approve of course)

Note that if you are only making changes in spack-stack (without changes in spack) then you don't need to go through this process because in this case the spack submodule commit hash remains the same.

Comment (climbfuji) [reply .gitmodules]: I should save this writeup because it's so short and clear. Thanks, Steve!

Comment (ashley314) [reply .gitmodules]: Yes awesome explanation! Thanks for clarifying :)

**Review by ashley314 — APPROVED** [review]
Thanks Steve for walking me through this and the examples

**Review by climbfuji — APPROVED** [review]

**Pull request merged**
